instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/instrumentation.rb

```diff
@@ -10,7 +10,15 @@ module Mysql2
       # The Instrumentation class contains logic to detect and install the Mysql2
       # instrumentation
       class Instrumentation < OpenTelemetry::Instrumentation::Base
-        install do |_config|
+        install do |config|
+          if config[:enable_sql_obfuscation]
+            config[:db_statement] = :obfuscate
+            OpenTelemetry.logger.warn(
+              'Instrumentation mysql2 configuration option enable_sql_obfuscation has been deprecated,' \
+              'use db_statement option instead'
+            )
+          end
```

Comment:
Tim and I discussed this a bit on a call, I encouraged taking this approach where we would let the legacy option override the `db_statement` option if it was set.

The reason for this is that it is defensive towards someone who bumps the gem without checking the changelog.  If someone sets both, we'll assume they probably wanted to keep the obfuscation enabled and will get the warning.

We'll leave it in for a version or two and then remove the setting entirely.

Reply:
🤔 if they have the new option set, it most likely means that they added it explicitly and forgot to remove the old one. Anyway, this approach is the safest, so 👍

Reply:
Also, I guess we can't distinguish between "default value" and "explicitly set to default value"?

Reply:
Definitely focusing on safety here.

We should be able to distinguish between default and explicitly set, those changes would require pushing the deprecation support into the instrumentation base as you suggested above.

https://github.com/open-telemetry/opentelemetry-ruby/blob/9e86944a0347a63b393246cf94d585e04e17b539/instrumentation/base/lib/opentelemetry/instrumentation/base.rb#L267-L277